### PR TITLE
Remove: Knex Cli as Global Deep

### PR DESCRIPTION
**Pull request by guidiego**
I was talking with @ErisDS today, and talk about create a script for *db init* to use `knex-migrator` as local deep! Here a little solution :metal:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,7 +60,7 @@
     "intl-messageformat": "1.3.0",
     "jsonpath": "0.2.9",
     "knex": "0.12.5",
-    "knex-migrator": "2.0.7",
+    "knex-migrator": "^2.0.7",
```
Comment (acburdine) [inline package.json]: Could this change be removed please? We like to keep our dependencies pinned at a specific version. Thanks!

**Review by acburdine — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (guidiego) [inline package.json]: Yeah, my bad! Updated ;)

**Comment by ErisDS** [comment]
This doesn't reflect the full range of options provided by knex-migrator. I'm happy to hear more opinions, but I think it's perfectly acceptable to require a global dependency here as it is for core contributors ONLY. People installing Ghost will need a different global dependency - `ghost-cli`.

We get complaints that the app is too monolithic, and now we've started splitting Ghost up we get complaints about global deps. We're literally never going to make everyone happy.

**Comment by guidiego** [comment]
If we are back into engineering stuff, everyone know about how evil can be Global state hahaha I still thinking that are not a good ideia put many deeps on root machines. In python we have `virtualenvs` to wrap our environments. Django, for a example, not have a cli but a `manege.py` with all cli stuff inside. In node our closest approach for isolate development environments is use power of `package.json` to organizate scripts or use Docker as well :)

But is nice to discuss a little about this, how you say, knex have many options. This can be provide using `yarn dbinit -- --flag`.

Thx for attention :metal:

**Pull request closed without merging**